src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -140,39 +117,32 @@ public static Logger getLogger() {
 
   public static void gameOverClearCache() {
     // Is static, set to null so that we don't keep the data around after a game is exited
-    s_battleCalculator.setGameData(null);
+    ProBattleUtils.setData(null);
     ProLogUI.clearCachedInstances();
   }
 
-  public IOddsCalculator getCalc() {
-    return s_battleCalculator;
-  }
-
-  @Override
-  public final GameData getGameData() {
-    if (data != null) {
-      return data;
-    } else {
-      return super.getGameData();
-    }
-  }
-
   @Override
   public void stopGame() {
     super.stopGame(); // absolutely MUST call super.stopGame() first
-    s_battleCalculator.cancel(); // cancel any current calcing
+    ProBattleUtils.cancel(); // cancel any current calcing
   }
 
   public void setStoredStrafingTerritories(final List<Territory> strafingTerritories) {
     storedStrafingTerritories = strafingTerritories;
   }
 
+  private void initializeData() {
```

Comment:
So I thought about this for a while. I think I'd rather have a static data class for high level constants rather than injecting an object into every class. I'm definitely more used to web programming so not sure if there is a better answer here but I feel like IOC isn't better than the current solution.